src/index.ts

```diff
@@ -226,81 +268,68 @@ export default class SwupScrollPlugin extends Plugin {
-		requestAnimationFrame(() => this.swup.scrollTo(top, this.shouldAnimate('betweenPages')));
-	};
-
-	/**
-	 * Stores the current scroll positions for the URL we just came from
-	 */
-	onBeforeReplaceContent = () => {
-		this.storeScrollPositions(this.currentCacheKey);
-		this.currentCacheKey = this.getCurrentCacheKey();
+		requestAnimationFrame(() => this.swup.scrollTo?.(top, this.shouldAnimate('betweenPages')));
 	};
 
 	/**
-	 * Deletes the scroll positions for the URL a link is pointing to,
-	 * if shouldResetScrollPosition evaluates to true
-	 * @returns {void}
+	 * Resets cached scroll positions for visits with a trigger element,
+	 * where shouldResetScrollPosition returns true for that trigger
 	 */
-	maybeResetScrollPositions(visit) {
+	maybeResetScrollPositions = (visit: Visit): void => {
 		const { url } = visit.to;
 		const { el } = visit.trigger;
-		const shouldReset = !el || this.options.shouldResetScrollPosition(el);
-		if (shouldReset) {
+
+		if (el && this.options.shouldResetScrollPosition(el)) {
 			this.resetScrollPositions(url);
 		}
-	}
+	};
 
 	/**
 	 * Stores the scroll positions for the current URL
-	 * @param {string} url
-	 * @returns {void}
 	 */
-	storeScrollPositions(url) {
+	cacheScrollPositions(url: string): void {
 		// retrieve the current scroll position for all containers
 		const containers = queryAll(this.options.scrollContainers).map((el) => ({
 			top: el.scrollTop,
 			left: el.scrollLeft
 		}));
 
 		// construct the final object entry, with the window scroll positions added
-		this.scrollPositionsStore[url] = {
+		const positions = {
 			window: { top: window.scrollY, left: window.scrollX },
 			containers
 		};
+
+		this.cachedScrollPositions[url] = positions;
 	}
 
 	/**
 	 * Resets stored scroll positions for a given URL
-	 * @param {string} url
 	 */
-	resetScrollPositions(url) {
+	resetScrollPositions(url: string): void {
 		const cacheKey = this.swup.resolveUrl(url);
-		delete this.scrollPositionsStore[cacheKey];
-		this.scrollPositionsStore[cacheKey] = null;
+		delete this.cachedScrollPositions[cacheKey];
 	}
 
 	/**
 	 * Get the stored scroll positions for a given URL from the cache
-	 * @returns {(object|undefined)}
 	 */
-	getStoredScrollPositions(url) {
+	getCachedScrollPositions(url: string): ScrollPositionsCacheEntry | undefined {
 		const cacheKey = this.swup.resolveUrl(url);
-		return this.scrollPositionsStore[cacheKey];
+		return this.cachedScrollPositions[cacheKey];
 	}
 
 	/**
 	 * Restore the scroll positions for all matching scrollContainers
-	 * @returns void
 	 */
-	restoreScrollContainers() {
+	restoreScrollContainers(url: string): void {
 		// get the stored scroll positions from the cache
-		const scrollPositions = this.getStoredScrollPositions(this.getCurrentCacheKey()) || {};
-		if (scrollPositions.containers == null) {
+		const scrollPositions = this.getCachedScrollPositions(url);
+		if (!scrollPositions || scrollPositions.containers.length === 0) {
```

Comment:
Playing code golf: `if (!scrollPositions?.containers?.length)` is a bit shorter, in case you also find it still readable.

Reply:
😄 There was some reason why i did it like this.

But feel free to push any changes to this PR!

Reply:
Let's leave it as-is if there was a reason to do it this way.

Reply:
The reason was that the check for missing scrollPositions prevents a possibly unnecessary `queryAll` further down in the function.